integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/detailquery/SearchModeWithShardSuite.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.spark.testsuite.detailquery
+
+import org.apache.carbondata.datamap.lucene.LuceneFineGrainDataMapSuite
+import org.apache.spark.sql.test.util.QueryTest
+import org.apache.spark.sql.{CarbonSession, Row}
+import org.scalatest.BeforeAndAfterAll
+
+/**
+  * Test Suite for search mode with shard
+  */
+
+class SearchModeWithShardSuite extends QueryTest with BeforeAndAfterAll {
+  val file = resourcesPath + "/datamap_input.csv"
+
+  override def beforeAll = {
+    sqlContext.sparkSession.asInstanceOf[CarbonSession].startSearchMode()
+    LuceneFineGrainDataMapSuite.createFile(file, 1000000)
+    sql("DROP TABLE IF EXISTS datamap_test_table")
+  }
+
+  override def afterAll = {
+    LuceneFineGrainDataMapSuite.deleteFile(file)
+    sql("DROP TABLE IF EXISTS datamap_test_table")
+    sqlContext.sparkSession.asInstanceOf[CarbonSession].stopSearchMode()
+  }
+
+  private def sparkSql(sql: String): Seq[Row] = {
+    sqlContext.sparkSession.asInstanceOf[CarbonSession].sparkSql(sql).collect()
+  }
+
+  test("test search mode with shard to search") {
```

Comment:
add this to existing test suite `SearchModeTestcase`

Reply:
ok, done